File: 02-sample-prompts/files/BankAccount.java
import java.util.Scanner;
 
public class BankAccount {
    private double balance;
 
    public BankAccount() {
        this.balance = 0;
    }
 
    public void deposit(double amount) {
        this.balance += amount;
    }
 
    public void withdraw(double amount) {
        if (amount <= this.balance) {
            this.balance -= amount;
        } else {
            System.out.println("Insufficient funds");
        }
    }
 
    public double getBalance() {
        return this.balance;
    }
 
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        BankAccount account = new BankAccount();
 
        while (true) {
            System.out.println("1. Deposit");
            System.out.println("2. Withdraw");
            System.out.println("3. Balance inquiry");
            System.out.println("4. Exit");
 
            int choice = scanner.nextInt();
            switch (choice) {
                case 1:
                    System.out.print("Enter amount to deposit: ");
                    double depositAmount = scanner.nextDouble();
                    account.deposit(depositAmount);
                    break;
                case 2:
                    System.out.print("Enter amount to withdraw: ");
                    double withdrawAmount = scanner.nextDouble();
                    account.withdraw(withdrawAmount);
                    break;
                case 3:
                    System.out.println("Current balance is " + account.getBalance());
                    break;
                case 4:
                    return;
                default:
                    System.out.println("Invalid choice");
            }
        }
    }
}

File: 02-sample-prompts/files/BankAccount.py
class BankAccount:

    def __init__(self):
        self.balance = 5000.0
        self.name = "demouser"
        self.address = "123 Main Street"
        self.accountnum = "1234567890"
        self.country = "India"

    def deposit(self, amount):
        self.balance += amount

    def withdraw(self, amount):
        if amount <= self.balance:
            self.balance -= amount
        else:
            print("Insufficient funds")

    def get_balance(self):
        return self.balance

def main():
    account = BankAccount()
    while True:
        print("1. Deposit")
        print("2. Withdraw")
        print("3. Balance inquiry")
        print("4. Exit")
        choice = int(input("Enter your choice: "))
        if choice == 1:
            amount = float(input("Enter the amount to deposit: "))
            account.deposit(amount)
        elif choice == 2:
            amount = float(input("Enter the amount to withdraw: "))
            account.withdraw(amount)
        elif choice == 3:
            balance = account.get_balance()
            print(f"Current balance is {balance}")
        elif choice == 4:
            break
        else:
            print("Invalid choice")

if __name__ == "__main__":
    main()
